packages/rum-core/src/common/apm-server.js

```diff
@@ -252,29 +226,47 @@ class ApmServer {
     })
   }
 
-  sendTransactions(transactions) {
-    if (this._configService.isValid() && this._configService.isActive()) {
-      if (transactions && transactions.length > 0) {
-        const payload = {
-          service: this.createServiceObject(),
-          transactions
-        }
-        const filteredPayload = this._configService.applyFilters(payload)
-        if (filteredPayload) {
-          const endPoint = this._configService.getEndpointUrl('transactions')
-          const ndjson = this.ndjsonTransactions(filteredPayload.transactions)
-          ndjson.unshift(
-            NDJSON.stringify({ metadata: { service: filteredPayload.service } })
-          )
-          const ndjsonPayload = ndjson.join('')
-          return this._postJson(endPoint, ndjsonPayload)
-        } else {
-          this._loggingService.warn('Dropped payload due to filtering!')
-        }
-      }
-    } else {
+  _send(data, type = 'transaction') {
```

Comment:
I don't think we should have a default value here, if `type` isn't provided we should either throw or return!

I think we should also check `data`!

Reply:
fixed as well.